Recurring Payments: Use Upgrade and Stripe Nudges

ockham opened the pull request:
**Fusion didn't work, so I generated the WP.com counterpart manually: D32570-code.**

Take Two of #13233, which was getting a bit noisy. This time, I'm also adding the Stripe Connect nudge since it's rather hard to do add each nudge in a separate PR, due to the block's UX. Thus, this PR is essentially implementing mockups from p8hgLy-1RC-p2 (with some limitations, see below).

The Stripe Nudge is used both on Jetpack and WP.com, wheras the Upgrade Nudge is only used when the relevant filter returns true (i.e. currently only on WP.com).

#### Changes proposed in this Pull Request:

Add an UpgradeNudge and a Stripe Connect Nudge to the Recurring Payments block to replace UI that's currently part of the block itself.

####  Screenshots

##### Before

![image](https://user-images.githubusercontent.com/96308/63064516-c4d9ee80-bf01-11e9-870d-7fd844dd7b17.png)

![image](https://user-images.githubusercontent.com/96308/64378912-099f0400-d02e-11e9-9f6a-369facfc4d47.png)

##### After

![image](https://user-images.githubusercontent.com/96308/64385879-ec6f3300-d037-11e9-86bd-0b2835b183b6.png)

![image](https://user-images.githubusercontent.com/96308/64388758-5db2e400-d040-11e9-9288-c551e1ec04be.png)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

The Recurring Payments block isn't new; we're just using the Upgrade Nudge, and introducing the Stripe Connect Nudge to replace some previously in-button messaging.

#### Testing instructions:

On Jetpack, verify that the Stripe Nudge is now shown if the site is on a paid plan, but not yet connected to Stripe; furthermore, the Recurring Payments button will be previewed inside the block in that case. The block's behavior should otherwise be unchanged.

- Switch to this branch locally, build Jetpack (`yarn build`), and start Docker (`yarn docker:up`)
- Make sure you're connected to WP.com, and on a free plan
- Start a new post, and insert the 'Recurring Payments' block
- Verify that it prompts you to upgrade to a paid plan (from within the block; no upgrade nudge)
- Upgrade to a paid plan
- Verify that post-checkout, you're redirected back to the block editor. You should now see the Upgrade Nudge, and an interactive preview of the Recurring Payments button inside of the block.

Test the WP.com counterpart:

- Apply D32570-code to your sandbox
- Make sure you're on a free plan
- Start a new post, and insert the 'Recurring Payments' block
- Verify that you can change the button's label and colors
- Click the nudge's "Upgrade" CTA button
- Pay for the premium plan
- Verify that you're redirected back to where you came from (the wp-admin block editor) (without any intermediate redirects on WP.com)
- Verify that you now see the Stripe Connect Nudge on top of the block
- Click the Connect button and connect to stripe (see the Field Guide for how to do this in sandbox mode)
- Manually return to the editor and verify that the nudges are gone

_You can also simulate the WP.com behavior in JP by adding `<?php add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );` to e.g. `docker/mu-plugins/0-gutenpack.php` and applying the following diff on top of this PR to your local JP Docker install:_

```diff
diff --git a/modules/memberships/class-jetpack-memberships.php b/modules/memberships/class-jetpack-memberships.php
index 52f9e92d2..e3e7b08ef 100644
--- a/modules/memberships/class-jetpack-memberships.php
+++ b/modules/memberships/class-jetpack-memberships.php
@@ -313,7 +313,7 @@ class Jetpack_Memberships {
 		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
 		// return statement below.
 		// phpcs:ignore
-		return Jetpack::is_active(); // && Jetpack_Plan::supports( 'recurring-payments' ); // phpcs:ignore
+		return Jetpack::is_active() && Jetpack_Plan::supports( 'recurring-payments' ); // phpcs:ignore
 	}
 
 	/**
```

#### Proposed changelog entry for your changes:

Add an Upgrade Nudge and a Stripe Connect Nudge to the Recurring Payments block to replace UI that's currently part of the block itself.

#### Differences from p8hgLy-1RC-p2

1. The Upgrade Nudge doesn't mention Stripe. For comparison, this is the mockup:
![rp-no-footer-4](https://user-images.githubusercontent.com/96308/64378497-21c25380-d02d-11e9-9e83-1d4e976f86e3.png)
The reason is that the Upgrade Nudge is currently generically added upon block registration. We'd have to move it to individual blocks in order to customize its message, but that will have us face another problem (similar to item 4. below).
2. In the Stripe Nudge message, there are no hyperlinks pointing to additional information. Again, see the mockup:
![connect-to-stripe-1](https://user-images.githubusercontent.com/96308/64378665-7960bf00-d02d-11e9-8a4d-2c66fe501b90.png)
The reason is that unlike `calypso-i18n`, it's not straight-forward to add hyperlinks to parts of a translated string within the framework of `@wordpress/i18n`. (There are a few ugly/fragile ways though.)
3. Post-checkout, the user continues to be redirected back to the editor (rather than showing them the success modal within Calypso). We've already added some logic both to Calypso and Jetpack to implement this behavior after some discussion. The feature hasn't been switched on in Jetpack yet, but once it is, it'll be hard to change the redirect behavior. We should make a very conscious decision here.
4. When the Stripe Nudge is shown, there's no border around the block (unless it becomes active by clicking into it). For the Upgrade Nudge, we add the filter by adding a class to the block container div that holds the block. This is done through a [filter](https://github.com/Automattic/jetpack/blob/005a488ce5de12631113665aa781fd6e2e7dd963/extensions/shared/register-jetpack-block.js#L53-L59) at block registration time since the block itself doesn't have direct access to the surrounding container. The same principle doesn't easily carry over to the Stripe Nudge, since unlike block availability and required plan, we only obtain information about Stripe connection status through a network request fired by the block.

I'd suggest we discuss each of these items separately to make a plan on how to proceed with them. I think that it's conceivable that we don't block this PR by them.

ockham commented:
Fusion didn't work, so I generated the WP.com counterpart manually: D32570-code.

sirreal commented on modules/memberships/class-jetpack-memberships.php:
```diff
@@ -277,5 +288,55 @@ public static function get_blog_id() {
+		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
+		// return statement below.
+		// phpcs:ignore
+		return Jetpack::is_active(); // && Jetpack_Plan::supports( 'recurring-payments' ); // phpcs:ignore
```
Here we're hitting the situation where the upgrade nudge is not an enhancement, but an alternative. We should be inspecting the filter version to know whether we should mark unavailable —in which case the nudge will be applied— or circumvent the nudge mechanism —in which case the original behavior is maintained.

We'll probably need to pull the filter check into a dedicated method to keep it's default in a single place:

```php
public static function is_upgrade_nudge_available() {
  return apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false );
}
```

Then we can check it here:

```suggestion
		return Jetpack::is_active() && (
			! Jetpack_Gutenberg::is_upgrade_nudge_available() ||
			Jetpack_Plan::supports( 'recurring-payments' )
		);
```

ockham replied:
We can do that; not sure how to best handle the [client side](https://github.com/Automattic/jetpack/pull/13394#discussion_r321776190) though.

I was cutting some corners here, since doing this "right" seemed to require a rather complex solution for what I still hope to be somewhat temporary (hoping to remove all the `this.isJetpack &&` stuff once the backend installer is sorted out and we'll be able to enable nudges in Jetpack).

sirreal replied:
Did you consider something like the suggestion above?

ockham replied:
Ahh, missed it, my bad. Makes a lot of sense, I'll apply it.

sirreal commented on extensions/blocks/recurring-payments/edit.jsx:
```diff
@@ -54,6 +55,16 @@ class MembershipsButtonEdit extends Component {
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, remove the line below, and replace all
+		// checks for `this.isJetpack` with `false`.
+		this.isJetpack = get( window, [ '_currentSiteType' ] ) !== 'simple';
```
I've left similar notes on the PHP side of this, but this isn't quite accurate. Whether or not the nudge is used depends on a filter. That filter may enable or disable the feature on Jetpack sites at any time, now or in the future.

I'm wondering what the best way to support the filter here is 🤔 

…🤔 What if the upgrade nudge wrapper passed a prop down to the component it wraps that indicates it's inside the nudge? I think that would give us the information we need here.

ockham replied:
> … What if the upgrade nudge wrapper passed a prop down to the component it wraps that indicates it's inside the nudge? I think that would give us the information we need here.

That wouldn't cover the Stripe Nudge case though, would it? :thinking:

ockham replied:
(Fixed to use a better solution, after a call with @sirreal. Thanks for your suggestion!)

sirreal reviewed (dismissed):
It seems that we should take the filter value into account rather than checking whether a site is a Jetpack site.

simison commented:
Some non-blocking visual issues:

Spacing in this area:

<img width="376" alt="Screenshot 2019-09-13 at 18 51 16" src="https://user-images.githubusercontent.com/87168/64876506-e1c82580-d657-11e9-864c-69319808416f.png">
<img width="526" alt="Screenshot 2019-09-13 at 18 50 37" src="https://user-images.githubusercontent.com/87168/64876507-e1c82580-d657-11e9-9880-18e4ed1fbf27.png">

When aligning the block, border on the left of upgrade nudge disappears:

<img width="717" alt="Screenshot 2019-09-13 at 18 52 06" src="https://user-images.githubusercontent.com/87168/64876532-ebea2400-d657-11e9-9f40-470ecac34f33.png">

simison commented:
I couldn't reproduce the issue but for little while I ended up directed to "plans" page right from the checkout, without having change to buy the plan.

Then I went to Jetpack plans page, pressed "upgrade to personal" and the checkout step didn't redirect me. I noticed the URL having two additional ARGs upgrade nudge button didn't have: `?site=<SITE>&u=1`

Now the upgrade nudge works without issues, too.

This was happening with totally fresh Jetpack Docker site, on free plan, connected as primary user.

simison commented:
Not necessarily in this PR, but I don't see any blockers to actually gate out the Stripe nudge. We could very well ship that in Jetpack even if we gate out the Plans nudge?

So we'd have this:

<img width="507" alt="image" src="https://user-images.githubusercontent.com/87168/64878747-e8a56700-d65c-11e9-8762-af988b36d5d9.png">


...instead of this:

<img width="441" alt="Screenshot 2019-09-13 at 19 27 09" src="https://user-images.githubusercontent.com/87168/64878625-99f7cd00-d65c-11e9-8327-bf0b77b8be12.png">

...at both .com and Jetpack.

ockham commented:
> Not necessarily in this PR, but I don't see any blockers to actually gate out the Stripe nudge. We could very well ship that in Jetpack even if we gate out the Plans nudge?
> 
> So we'd have this:
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/87168/64878747-e8a56700-d65c-11e9-8762-af988b36d5d9.png">
> 
> ...instead of this:
> 
> <img alt="Screenshot 2019-09-13 at 19 27 09" width="441" src="https://user-images.githubusercontent.com/87168/64878625-99f7cd00-d65c-11e9-8327-bf0b77b8be12.png">
> 
> ...at both .com and Jetpack.

That would probably work and allow us to remove some code. OTOH, it might be a bit weird (in JP) to start with in-block instructions (telling you to upgrade your plan), and then seeing a nudge outside the block? Probably not a dealbreaker, but a bit weird.

ockham commented:
> Some non-blocking visual issues:
> 
> Spacing in this area:
> 
> <img alt="Screenshot 2019-09-13 at 18 51 16" width="376" src="https://user-images.githubusercontent.com/87168/64876506-e1c82580-d657-11e9-864c-69319808416f.png">
> 
> <img alt="Screenshot 2019-09-13 at 18 50 37" width="526" src="https://user-images.githubusercontent.com/87168/64876507-e1c82580-d657-11e9-9880-18e4ed1fbf27.png">

Just spent an hour trying to fix this :roll_eyes: Did I mention that I hate CSS?

ockham commented:
> Just spent an hour trying to fix this  Did I mention that I hate CSS?

Oh actually 5c990a0 :grimacing:

jancavan commented:
> That would probably work and allow us to remove some code. OTOH, it might be a bit weird (in JP) to start with in-block instructions (telling you to upgrade your plan), and then seeing a nudge outside the block? Probably not a dealbreaker, but a bit weird.

Not sure if I'm understanding correctly, but how does this work? Upgrade nudge is in-block to start, meaning, they also won't be able to interact with the button. However, after upgrade completion, they can finally interact with the button and a Stripe nudge appears at the top of the block?

ockham committed:
Recurring Payments: Use Upgrade and Stripe Nudges

ockham committed:
Add tracking

ockham commented:
> > That would probably work and allow us to remove some code. OTOH, it might be a bit weird (in JP) to start with in-block instructions (telling you to upgrade your plan), and then seeing a nudge outside the block? Probably not a dealbreaker, but a bit weird.
> 
> Not sure if I'm understanding correctly, but how does this work? Upgrade nudge is in-block to start, meaning, they also won't be able to interact with the button. However, after upgrade completion, they can finally interact with the button and a Stripe nudge appears at the top of the block?

Yep, that's what @simison suggested, if I understood him correctly.

I also just had a call with @sirreal -- it seems that people agree that we should always show the Stripe nudge. It's certainly going to make the code quite a bit leaner, so I'll proceed and implement that.

ockham committed:
Show the Stripe Nudge on both JP and WP.com

ockham commented:
This PR should be ready for another look. I'll update the WP.com side diff now.

sirreal reviewed (commented):
Great to see this becoming simpler 👍

ockham commented:
> This PR should be ready for another look. I'll update the WP.com side diff now.

Done :sweat_smile:

ockham commented on extensions/blocks/recurring-payments/edit.jsx:
```diff
@@ -56,6 +58,19 @@ class MembershipsButtonEdit extends Component {
+		// This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, remove the line below, and replace all
+		// checks for `this.hasUpgradeNudge` with `false`.
```
Oops, I think I forgot to update that last sentence. I don't think that's accurate anymore :sweat_smile:

sirreal replied:
Right, I think we rely on the nudge being enabled or disabled by the filter. We should be able to remove this entire comment.

ockham committed:
RecPayments: Block is available on Free plan if nudge is disabled

Co-Authored-By: Jon Surrell <jon.surrell@automattic.com>

ockham commented on modules/memberships/class-jetpack-memberships.php:
```diff
@@ -277,5 +288,58 @@ public static function get_blog_id() {
+		// connect Stripe. This however means that we have to display the block also if the site's on a Free plan.
+		//
+		// Check p7rd6c-23C-p2 for details and progress. If ready, uncomment the right-hand side of the line with the
+		// return statement below.
```
We should update this comment to reflect the change on L317, [see](https://github.com/Automattic/jetpack/pull/13394#discussion_r321672984)

jancavan commented:
> Verify that it prompts you to upgrade to a paid plan (from within the block; no upgrade nudge)
> Upgrade to a paid plan
> Verify that post-checkout, you're redirected back to the block editor. You should now see the Upgrade Nudge, and an interactive preview of the Recurring Payments button inside of the block.

On the Jetpack testing instructions, not sure why I should be seeing the Upgrade Nudge again post-checkout? Shouldn't it be the Stripe nudge instead?

I can't confirm that I'm being redirected back to the block editor. I'm seeing these two screens after upgrading:

![Screenshot 2019-09-24 14 20 29](https://user-images.githubusercontent.com/4924246/65551339-8d823880-ded6-11e9-96b6-3094a2af7584.png)

![Screenshot 2019-09-24 14 20 21](https://user-images.githubusercontent.com/4924246/65551341-907d2900-ded6-11e9-88ee-43217456afad.png)

I'm seeing the Stripe nudge now though and the RP block is now interactive. How do I modify the actual payment settings?

After Stripe connection has been established, I'm now seeing the gray block again re-appearing above the interactive button:

![Screenshot 2019-09-24 14 27 26](https://user-images.githubusercontent.com/4924246/65551771-8576c880-ded7-11e9-9393-be387ae65401.png)

jancavan commented:
I honestly couldn't figure out how to disconnect my Stripe account from wpcom, so I went ahead and revoked its access from my Stripe dashboard.

Upon refreshing the editor, I was expecting to see the Stripe nudge again, but I couldn't get it to display again. The page preview also doesn't show the RP block that I added.

I think disconnecting Stripe should allow you to reconnect again from the editor - ie: Stripe nudge should show up again.

jancavan commented:
> In the Stripe Nudge message, there are no hyperlinks pointing to additional information.

If we couldn't do this, we need to change the verbiage, otherwise it isn't actionable:

_Connect to Stripe to use this block on your site
This block will be hidden from your visitors until you connect to Stripe. [ Connect ]_

I think the Upgrade Nudge is also missing the word "your":

_Upgrade to Jetpack Personal to use this block on your site
You can try it out before upgrading, but only you will see it. It will be hidden from **_your_** visitors until you upgrade. [ Upgrade ]_

sirreal commented on modules/memberships/class-jetpack-memberships.php:
```diff
@@ -277,5 +288,58 @@ public static function get_blog_id() {
+		// return statement below.
+		// phpcs:ignore
+		return Jetpack::is_active() && (
+			! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ) || // Remove when the default becomes `true`
```
Would be good to add a comment pointing here in `Jetpack_Gutenberg` at the canonical location of this filter:

https://github.com/Automattic/jetpack/blob/21347595efacb02335f24bf3af236573c2211fef/class.jetpack-gutenberg.php#L202

simison replied:
Good spot; I'll add the standard `/** This filter is documented in class.jetpack-gutenberg.php */` banner

simison replied:
added in https://github.com/Automattic/jetpack/pull/13394/commits/8ba084f826ea78bd30bc900c76af7a1215c2a13f

sirreal replied:
I really meant that that filter _there_ should have a note to remove this line when the default value is updated.

simison replied:
Gotcha 👍  added https://github.com/Automattic/jetpack/pull/13394/commits/19e8af690877a8d864047a04e89383ead61c1384

simison committed:
Refer to `class.jetpack-gutenberg.php` for filter's documentation

simison committed:
Update copy in stripe and upgrade banners

simison commented:
Updated the copy to what @jancavan suggested. There is an experiment in another pr (https://github.com/Automattic/jetpack/pull/13527) to support HTML in these texts but let's leave that for another PR.

simison committed:
Mention removing filter from class-jetpack-memberships.ph once default value is changed

simison reviewed (dismissed):
Tests well for Jetpack sites.

I'm testing and commenting the .com aspects separately in D32570-code

sirreal committed:
Remove redundant comments

sirreal commented:
Rebased.

sirreal reviewed (approved):
This is testing well for Jetpack. Note that it's pesky to test because you have to delete the site option from Jetpack and WPCOM in order to retest the Stripe connection nudge.

sirreal commented:
Some of the product creation (recurring payment plans) seems to behave strangely. I'll investigate.

sirreal reviewed (dismissed):
My issues seem to have been related to sandboxing and when I got them fixed, this worked as expected. I think this is ready to ship!

sirreal commented:
There does seem to be a regression where the block does not render on the frontend. I'm debugging now.

sirreal commented on extensions/blocks/recurring-payments/recurring-payments.php:
```diff
@@ -9,11 +9,5 @@
-			'render_callback' => array( Jetpack_Memberships::get_instance(), 'render_button' ),
-		)
-	);
+	Jetpack_Memberships::get_instance();
```
This isn't working on the frontend 😞 

Instead of a direct call to register the block, this moves it behind an action:
https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/modules/memberships/class-jetpack-memberships.php#L70-L73
https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/modules/memberships/class-jetpack-memberships.php#L104

That action runs here:

https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/class.jetpack-gutenberg.php#L370-L386

Which is before this file is included and the action is registered.

<details>

<summary>Some relevant places in code</summary>

https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/class.jetpack-gutenberg.php#L237-L271
https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/class.jetpack.php#L613
https://github.com/Automattic/jetpack/blob/11088101cea4ef93ab1e2eba00a24cbdecc486ff/class.jetpack-gutenberg.php#L660-L673

</details>

sirreal replied:
Addressed this in 682fce9

sirreal reviewed (dismissed):
This regressed the block on the frontend, it does not render. We'll need to revisit it to detangle the flow.

sirreal committed:
Ensure registration occurs from the extensions php

This fixes a regression where the block was not registered when viewed
on the frontend.

sirreal reviewed (dismissed):
I've been through a lot of testing and this seems to be working well in all cases now.

sirreal committed:
Organize class variables

sirreal committed:
Ensure wpcom is wpcom

sirreal committed:
[not verified] Revert "Ensure wpcom is wpcom"

This reverts commit 12d1fda07c79795f87bdcc68da4c6f751f9e283a.

jeherve commented:
I am trying to test this, but can’t seem to see the nudge when on a free plan. It seems to happen both locally and on a JN site. I have the `jetpack_block_editor_enable_upgrade_nudge` filter set, and I can see the nudge for the Simple Payments block. The Recurring Payments block, however, does not change.

sirreal committed:
Ensure upgrade and stripe nudge don't appear together

simison commented:
.com side deployed in r197188-wpcom

sirreal commented:
> .com side deployed in r197188-wpcom
This was reverted and we'll handle this in multiple deploys.

simison commented:
Deployed to .com in:

r197233-wpcom
r197234-wpcom
r197236-wpcom
r197239-wpcom
r197243-wpcom

Boom!

simison reviewed (dismissed):
Tests well in Jetpack:

<img width="760" alt="Screenshot 2019-10-02 at 11 16 19" src="https://user-images.githubusercontent.com/87168/66028888-b45bf280-e506-11e9-8a21-1b3db8a68518.png">


Tested with and without `add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );`

Will need some work later on when blocks internal upgrade nudge can be replaced with the banner also in Jetpack plugin:

<img width="743" alt="Screenshot 2019-10-02 at 11 19 24" src="https://user-images.githubusercontent.com/87168/66028874-af973e80-e506-11e9-8beb-18dbf1cf47ce.png">

simison commented:
Stripe nudge will need some updates to color variables if https://github.com/Automattic/jetpack/pull/13503 lands before this.

simison committed:
Fix white after reabasing with calypso-build v4 update

jeherve reviewed (commented):
I don't really understand why we have this "Add Text..." button that I can't modify, and that doesn't match the different options I'll have when Stripe is connected?

![image](https://user-images.githubusercontent.com/426388/66030788-60e7a580-e502-11e9-82f8-ba90c7b8930d.png)

Once I've connected Stripe, the button remains, but with an add margin.

![image](https://user-images.githubusercontent.com/426388/66031001-d2bfef00-e502-11e9-96fc-d5196b4dd622.png)

It only disappears when I add a real button:

![image](https://user-images.githubusercontent.com/426388/66031053-e8351900-e502-11e9-8ad6-8c8ffb566cbb.png)

simison committed:
Use primary blue instead of wpcom blue

For some reason, --color-wpcom didn't work (wasn't found). `--color-blue` does work and feels anyway better, less wpcom specific variable here.

Both should be `$muriel-blue-500` anyway.

simison commented:
>I don't really understand why we have this "Add Text..." button that I can't modify, and that doesn't match the different options I'll have when Stripe is connected?

True, while the placeholder button lets you play with the button's appearance (including text) it can be confusing. It's still better than it was before when there was no indication of any kind how the block might look like after plan purchase + stripe connection, so I'd say not a blocker here and good for a follow up some another time.

simison commented:
I rebased this PR with Calypso-build v4 update (https://github.com/Automattic/jetpack/pull/13503) and fixed Stripe star colors that broke in the update:

<img width="315" alt="image" src="https://user-images.githubusercontent.com/87168/66031382-f5a2d100-e50b-11e9-89ad-a531ccab35bb.png">

jeherve commented:
Ok. It does look off to me, so I'd definitely vote for a follow-up! Other than that, everything looks good to me.

simison commented:
>definitely vote for a follow-up! Other than that, everything looks good to me.

I'll think about it and see if there's a creative easy win. :-)

Thanks!

simison commented:
Synced Stripe color modifications in r197337-wpcom
